### ISSUE TITLE
Fails to load solar_zenith_angle from SLSTR l1b data

### ISSUE THREAD

**Issue by TAlonglong** [pytroll/satpy#1013]
I'm trying to load the solar_zenith_angle as defined in the slstr_l1b yaml file but this fail with 

```
Traceback (most recent call last):
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/yaml_reader.py", line 771, in get_dataset_key
    return get_key(key, self.available_ids.keys(), **kwargs)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/__init__.py", line 249, in get_key
    raise KeyError("No dataset matching '{}' found".format(str(key)))
KeyError: "No dataset matching 'DatasetID(name='longitude', wavelength=None, resolution=1000, polarization=None, calibration=None, level=None, modifiers=())' found"
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/software/conda-pytroll/scripts/process-slstr.py", line 48, in <module>
    ql.load(['solar_zenith_angle'])
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/scene.py", line 967, in load
    self.read(**kwargs)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/scene.py", line 875, in read
    return self._read_datasets(nodes, **kwargs)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/scene.py", line 716, in _read_datasets
    new_datasets = reader_instance.load(ds_ids, **kwargs)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/yaml_reader.py", line 786, in load
    coordinates = self._get_coordinates_for_dataset_keys(dsids)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/yaml_reader.py", line 653, in _get_coordinates_for_dataset_keys
    cids = self._get_coordinates_for_dataset_key(dsid)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/yaml_reader.py", line 645, in _get_coordinates_for_dataset_key
    cids.append(self.get_dataset_key(cid))
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/yaml_reader.py", line 773, in get_dataset_key
    return get_key(key, self.all_ids.keys(), **kwargs)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/__init__.py", line 249, in get_key
    raise KeyError("No dataset matching '{}' found".format(str(key)))
KeyError: "No dataset matching 'DatasetID(name='longitude', wavelength=None, resolution=1000, polarization=None, calibration=None, level=None, modifiers=())' found"
```

```python
from satpy import Scene, find_files_and_readers

from satpy.utils import debug_on
debug_on()

files = find_files_and_readers(sensor='slstr',
                               base_dir='/data/pytroll/slstr/20191211-035810/',
                               reader='slstr_l1b')

ql = Scene(filenames=files)

ql.load(['solar_zenith_angle'])
```

@mraspaud suggested to replace the `longitude` with `longitude_ao` in the slstr_l1b.yaml for the solar_zenith_angle. I tested this, but it failes with similar messages:

```
Traceback (most recent call last):
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/yaml_reader.py", line 771, in get_dataset_key
    return get_key(key, self.available_ids.keys(), **kwargs)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/__init__.py", line 249, in get_key
    raise KeyError("No dataset matching '{}' found".format(str(key)))
KeyError: "No dataset matching 'DatasetID(name='longitude_ao', wavelength=None, resolution=1000, polarization=None, calibration=None, level=None, modifiers=())' found"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/software/conda-pytroll/scripts/process-slstr.py", line 48, in <module>
    ql.load(['solar_zenith_angle'])
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/scene.py", line 967, in load
    self.read(**kwargs)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/scene.py", line 875, in read
    return self._read_datasets(nodes, **kwargs)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/scene.py", line 716, in _read_datasets
    new_datasets = reader_instance.load(ds_ids, **kwargs)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/yaml_reader.py", line 786, in load
    coordinates = self._get_coordinates_for_dataset_keys(dsids)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/yaml_reader.py", line 653, in _get_coordinates_for_dataset_keys
    cids = self._get_coordinates_for_dataset_key(dsid)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/yaml_reader.py", line 645, in _get_coordinates_for_dataset_key
    cids.append(self.get_dataset_key(cid))
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/yaml_reader.py", line 773, in get_dataset_key
    return get_key(key, self.all_ids.keys(), **kwargs)
  File "/software/conda-pytroll/lib/python3.7/site-packages/satpy/readers/__init__.py", line 249, in get_key
    raise KeyError("No dataset matching '{}' found".format(str(key)))
KeyError: "No dataset matching 'DatasetID(name='longitude_ao', wavelength=None, resolution=1000, polarization=None, calibration=None, level=None, modifiers=())' found"
```
**Environment Info:**
 - OS: Linux
 - Satpy Version: 0.16.1

**Comment by TAlonglong** [comment]
I think the problem is that the reader cant figure out which resolution to use for the solar_zenith_angle. The resolution config says 1000, but the angles file definition only parse the files for `view`. I guess in my case the view=o comes first and there the latitude_an is not given and then it fails.

**Comment by mraspaud** [comment]
In principle this should be fixed in satpy 0.20.0, see #1082. Feel free to reopen if it's not the case.
